cloud/services/container/clusters/reconcile.go

```diff
@@ -99,18 +101,21 @@ func (s *Service) Reconcile(ctx context.Context) (ctrl.Result, error) {
 		conditions.MarkFalse(s.scope.ConditionSetter(), clusterv1.ReadyCondition, infrav1exp.GKEControlPlaneCreatingReason, clusterv1.ConditionSeverityInfo, "")
 		conditions.MarkFalse(s.scope.ConditionSetter(), infrav1exp.GKEControlPlaneReadyCondition, infrav1exp.GKEControlPlaneCreatingReason, clusterv1.ConditionSeverityInfo, "")
 		conditions.MarkTrue(s.scope.ConditionSetter(), infrav1exp.GKEControlPlaneCreatingCondition)
+		s.scope.GCPManagedControlPlane.Status.Initialized = true
```

Comment:
In the "creating" scenario we should wait until the kubeconfig has been generated to set Initialized to true.

Reply:
Updated. Now we are setting `initialized` to  true after the kubeconfig is generated.